src/url-source.cpp

```diff
@@ -326,40 +326,75 @@ void curl_loop(struct url_source_data *usd)
 										["content-type"];
 							}
 						} else {
+							try {
+								// Use Inja to render the template
+								nlohmann::json data;
+								if (response.body_parts_parsed
+									    .size() > 1) {
+									for (size_t i = 0;
+									     i <
+									     response.body_parts_parsed
+										     .size();
+									     i++) {
+										data["output" +
+										     std::to_string(
+											     i)] =
+											response.body_parts_parsed
+												[i];
+									}
+									// in "output" add an array of all the outputs
+									data["output"] =
+										response.body_parts_parsed;
+								} else {
+									data["output"] =
+										response.body_parts_parsed
+											[0];
+								}
```

Comment:
this block repeats below precisely the same - can we extract a function and not repeat? 🙏